### PR TITLE
removes the encryption provider

### PR DESCRIPTION
**Pull request by p0lyn0mial**
In 4.7 we switched CAO to manage its own encryption configuration. 

That means we can replace a dynamic provider with one that maintains a static list.

### CODE REVIEW

File: pkg/operator/starter.go [hunk 1]
```diff
@@ -438,6 +438,18 @@ func prepareOauthOperator(controllerContext *controllercmd.ControllerContext, op
 	return nil
 }
 
+type encryptionProvider []schema.GroupResource
```
Comment (p0lyn0mial) [inline pkg/operator/starter.go:441]: /hold

I'm going to move it to library-go.

A static provider will be used by oas, cao and kas-o

Comment (p0lyn0mial) [reply pkg/operator/starter.go:441]: https://github.com/openshift/library-go/pull/992

Comment (p0lyn0mial) [reply pkg/operator/starter.go]: merged.

/hold cancel

File: pkg/operator/starter.go [hunk 2]
```diff
@@ -28,12 +28,29 @@ import (
...
+	oauthapiconfigobservercontroller "github.com/openshift/cluster-authentication-operator/pkg/operator/configobservation"
+	"github.com/openshift/cluster-authentication-operator/pkg/operator/revisionclient"
+	"github.com/openshift/cluster-authentication-operator/pkg/operator/workload"
 	"github.com/openshift/library-go/pkg/authentication/bootstrapauthenticator"
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
 	libgoassets "github.com/openshift/library-go/pkg/operator/apiserver/audit"
 	workloadcontroller "github.com/openshift/library-go/pkg/operator/apiserver/controller/workload"
 	apiservercontrollerset "github.com/openshift/library-go/pkg/operator/apiserver/controllerset"
 	libgoetcd "github.com/openshift/library-go/pkg/operator/configobserver/etcd"
+	"github.com/openshift/library-go/pkg/operator/encryption"
```
Comment (p0lyn0mial) [inline pkg/operator/starter.go:53]: this is new, the rest was moved up.

Comment (stlaz) [reply pkg/operator/starter.go:53]: nit: please keep the local deps in a separate block

**Review by stlaz — CHANGES_REQUESTED** [review]

File: pkg/operator/starter.go (hunk 2 above)
Comment (p0lyn0mial) [reply pkg/operator/starter.go:53]: it will be hard to keep track since the local deps match "github.com/opeshisft" prefix

Comment (p0lyn0mial) [reply pkg/operator/starter.go:37]: anyway, I moved them to the bottom of the import statement :)

**Pull request merged**